components/system/components/Tag.js

```diff
@@ -0,0 +1,581 @@
+import * as React from "react";
```

Comment:
Also, there should be restrictions on what type of tags can be added. You should check/modify them both while typing and before adding them. Tags should be converted to lowercase, and only contain letters, numbers, or dashes. No spaces or unusual characters. 

If someone tries to type a non allowed character, they shouldn't be allowed to type it into the input (nothing will show up)
And if they type an uppercase character it'll be automatically converted to lowercase
And a space will automatically be converted to a dash

You can achieve this by controlling the input in the onChange function for the input

Reply:
Note: edited to change from hyphens to spaces. Spoke to haris and we decided that spaces look stylistically better than hyphens in tags